bundles/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/device/DimmerDevice.java

```diff
@@ -63,6 +65,10 @@ private boolean handleBrightnessChannel(ChannelUID channelUid, Command command)
 
             // Don't send value 0 as brightness value as it will give an error from the device.
             if (percentCommand.intValue() > 0) {
+                if (lastBrightnessState.equals(PercentType.ZERO)) {
+                    // if the light was off before, turn it on before setting the brightness value
+                    setOnOffState(channelUid, OnOffType.ON);
```

Comment:
I would say sending ON should always be done, because theoretical the state in the binding might not reflect the actual state.So if the device is off, but the binding still thinks it's on it will not be set on.
Also the ON should be send after the setting brightness. Because brightness might be 100, while setting it to 10 might cause a light flashing. This because brightness is kept separate as the on state on the device.

Reply:
Done. Also fixed the tests